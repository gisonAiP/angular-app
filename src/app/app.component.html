<div>
  <app-header></app-header>
  <div class="container-fluid">
    <div class="row pl-2 pr-1">
      <div class="col-md-2"></div>
      <div class="col-12 col-md-8 px-md-5">
        <app-search></app-search>
        <div class="row">
          <p class="pl-sx-4 sub-header font-weight-bold">CATEGORIES</p>
        </div>
        <app-categories></app-categories>
        <div class="row">
          <p class="sub-header font-weight-bold pt-2">ALL PRODUCTS</p>
        </div>
        <app-products></app-products>
      </div>
      <div class="col-lg-2"></div>
    </div>
  </div>
  <div class="mobile-add-button d-flex d-md-none">
    <img src="./assets/svg/action-button.svg" width="44"/>
  </div>
  <app-footer></app-footer>
</div>
