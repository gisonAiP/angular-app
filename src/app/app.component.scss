/* devanagari */
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/poppins/v20/pxiByp8kv8JHgFVrLDz8Z11lFc-K.woff2) format('woff2');
  unicode-range: U+0900-097F, U+1CD0-1CF6, U+1CF8-1CF9, U+200C-200D, U+20A8, U+20B9, U+25CC, U+A830-A839, U+A8E0-A8FB;
}
/* latin-ext */
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/poppins/v20/pxiByp8kv8JHgFVrLDz8Z1JlFc-K.woff2) format('woff2');
  unicode-range: U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
}
/* latin */
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/poppins/v20/pxiByp8kv8JHgFVrLDz8Z1xlFQ.woff2) format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}

* {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 400;
}

html, body {
  width: 100%;
  height: 100%;
  margin: 0;
}

.container-fluid {
  max-width: 1180px;
}

.sub-header {
  color: #767676;
  font-size: 10px;
}

.mobile-add-button {
  position: fixed;
  bottom: 100px;
  right: 30px;
  z-index: 1;
}
/*Mozilla Firefox*/
:root, .categories {
  scrollbar-width: thin;
  scrollbar-color: #dedcdc transparent;
}

/*Chrome*/
::-webkit-scrollbar {
  width: 10px;
  height: 10px;
}

::-webkit-scrollbar-track {
  background-color: #F9FAF9;
}

::-webkit-scrollbar-thumb {
  border-radius: 15px;
  box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.2);
}
