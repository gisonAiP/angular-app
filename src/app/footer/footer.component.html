<footer data-test="footer" class="d-flex d-md-none justify-content-around align-items-center py-2">
  <div class="text-center">
    <img src="./assets/svg/home.svg" width="22"/>
    <p class="m-0">Home</p>
  </div>
  <div class="text-center">
    <img src="./assets/svg/track.svg" width="22"/>
    <p class="m-0">Trip</p>
  </div>
  <div class="text-center">
    <img src="./assets/svg/compass.svg" width="22"/>
    <p class="m-0">Explore</p>
  </div>
  <div class="text-center">
    <img src="./assets/svg/repeat.svg" width="22"/>
    <p class="m-0">Marketplace</p>
  </div>
  <div class="text-center">
    <img src="./assets/svg/menu.svg" width="22"/>
    <p class="m-0">Menu</p>
  </div>
</footer>
