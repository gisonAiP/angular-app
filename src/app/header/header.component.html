<header data-test="header">
  <div class="header-desktop d-none d-md-flex">
    <div class="header-logo">
      <img src="./assets/svg/tripico.svg" alt="logo" height="33"/>
    </div>
    <input class="header-search" placeholder="Where are you going?"/>
    <div class="header-icons-container">
      <img src="./assets/svg/home.svg" alt="home" height="20"/>
      <img src="./assets/svg/briefcase.svg" alt="briefcase" height="20"/>
      <img src="./assets/svg/compass.svg" alt="compass" height="20"/>
      <img src="./assets/svg/repeat.svg" alt="repeat" height="20"/>
      <img src="./assets/svg/avatar.svg" alt="avatar" height="20"/>
    </div>
  </div>
  <div class="header-mobile d-flex d-md-none justify-content-between py-3 px-2">
    <div>
      <p class="title">Marketplace</p>
    </div>
    <div>
      <img src="./assets/svg/search2.svg" alt="search" height="30" class="px-1"/>
      <img src="./assets/svg/user.svg" alt="user" height="30" class="px-1"/>
      <img src="./assets/svg/bookmark.svg" alt="bookmark" height="30" class="px-1"/>
    </div>
  </div>
</header>
