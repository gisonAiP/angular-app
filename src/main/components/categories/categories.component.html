<div class="d-flex categories mr-md-5">
  <ng-container *ngFor="let category of categories$ | async">
    <app-category data-test="category" [title]="category.name" [imageURL]="category?.displayResources?.picUrl || './assets/svg/category.svg'"></app-category>
  </ng-container>
</div>
