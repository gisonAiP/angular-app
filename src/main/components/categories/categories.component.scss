.categories {
  flex-wrap: nowrap;
  overflow-x: scroll;
  margin-right: 10px;
}
