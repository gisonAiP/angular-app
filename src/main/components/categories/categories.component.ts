import { Component, OnInit } from '@angular/core';
import { MainFacade } from '../../state/main.facade';
import {Category} from "../../../shared/model/category.model";
import {Observable} from "rxjs";

@Component({
  selector: 'app-categories',
  templateUrl: './categories.component.html',
  styleUrls: ['./categories.component.scss']
})
export class CategoriesComponent implements OnInit {
  categories$: Observable<Category[]> = this.facade.categories$;
  constructor(private facade: MainFacade) { }

  ngOnInit(): void {
    this.facade.getCategories();
  }

}
