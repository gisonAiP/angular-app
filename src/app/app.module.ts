import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { HeaderComponent } from './header/header.component';
import { FooterComponent } from './footer/footer.component';
import { SharedModule } from '../shared/shared.module';
import { MainModule } from '../main/main.module';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { HttpClientModule } from '@angular/common/http';
import { StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { environment } from '../environments/environment';
import { StoreRouterConnectingModule } from '@ngrx/router-store';
import { mainReducer } from '../main/state/main.reducer';
import {EffectsModule} from "@ngrx/effects";
import {MainEffects} from "../main/state/main.effects";
import {RouterModule} from "@angular/router";
import {AppRoutingModule} from "./app-routing.module";

@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    FooterComponent,
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    AppRoutingModule,
    EffectsModule.forRoot([]),
    StoreModule.forRoot({
      // main: mainReducer,
    }),
    StoreDevtoolsModule.instrument({
      maxAge: 25, // Retains last 25 states
      logOnly: environment.production, // Restrict extension to log-only mode
    }),
    StoreRouterConnectingModule.forRoot(),
    MainModule,
    SharedModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
