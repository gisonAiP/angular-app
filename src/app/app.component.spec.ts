import { createRoutingFactory, Spectator } from '@ngneat/spectator/jest';
import { MockComponent } from 'ng-mocks';
import { AppComponent } from './app.component';
import { FooterComponent } from './footer/footer.component';
import { ProductsComponent } from '../main/components/products/products.component';
import { CategoriesComponent } from '../main/components/categories/categories.component';
import { HeaderComponent } from './header/header.component';

describe('AppComponent', () => {
  let spectator: Spectator<AppComponent>;
  const createComponent = createRoutingFactory({
    component: AppComponent,
    shallow: true,
    providers: [],
    declarations: [
      MockComponent(HeaderComponent),
      MockComponent(FooterComponent),
      MockComponent(ProductsComponent),
      MockComponent(CategoriesComponent),
    ]
  });

  beforeEach(() => {
    spectator = createComponent();
    spectator.detectChanges();
  });

  it('should validate snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});

