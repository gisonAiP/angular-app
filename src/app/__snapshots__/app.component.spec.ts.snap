// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should validate snapshot 1`] = `
<app-root
  title={[Function String]}
>
  <div>
    <app-header />
    <div
      class="container-fluid"
    >
      <div
        class="row pl-2 pr-1"
      >
        <div
          class="col-md-2"
        />
        <div
          class="col-12 col-md-8 px-md-5"
        >
          <app-search />
          <div
            class="row"
          >
            <p
              class="pl-sx-4 sub-header font-weight-bold"
            >
              CATEGORIES
            </p>
          </div>
          <app-categories />
          <div
            class="row"
          >
            <p
              class="sub-header font-weight-bold pt-2"
            >
              ALL PRODUCTS
            </p>
          </div>
          <app-products />
        </div>
        <div
          class="col-lg-2"
        />
      </div>
    </div>
    <div
      class="mobile-add-button d-flex d-md-none"
    >
      <img
        src="./assets/svg/action-button.svg"
        width="44"
      />
    </div>
    <app-footer />
  </div>
</app-root>
`;
